src/component/reducer/actions/DomainActions.ts

```diff
@@ -38,33 +38,31 @@ function getActiveData(draft: Draft<State>): Array<Datum1D> {
 function getDomain(draft: Draft<State>) {
   let xArray: Array<number> = [];
   let yArray: Array<number> = [];
-  let yDomains = {};
-  let xDomains = {};
+  let yDomains: Record<string, number[]> = {};
+  let xDomains: Record<string, number[]> = {};
 
   const data = getActiveData(draft);
   try {
-    xArray = data.reduce<Array<number>>((acc, d: Datum1D) => {
+    data.forEach((d: Datum1D) => {
       const { display, data } = d;
       const domain = [data.x[0], data.x[data.x.length - 1]];
       xDomains[d.id] = domain;
       if (display.isVisible) {
-        acc = acc.concat(domain);
+        xArray = xArray.concat(domain);
       }
-      return acc;
-    }, []);
+    });
 
-    yArray = data.reduce<Array<number>>((acc, d: Datum1D) => {
+    data.forEach((d: Datum1D) => {
       const { display } = d;
       const data = get1DDataXY(d);
 
       const _extent = extent(data.y) as Array<number>;
       yDomains[d.id] = _extent;
 
       if (display.isVisible) {
-        acc = acc.concat(_extent);
+        yArray = yArray.concat(_extent);
       }
-      return acc;
-    }, []);
+    });
```

Comment:
`data.filter(d => d.display.isVisible).map(() => extent).flat()`

Reply:
I think it's no possible this way because we have to change `yDomains[d.id] = _extent;`